homeassistant/components/sensor/eight_sleep.py

```diff
@@ -255,15 +269,17 @@ def state(self):
         """Return the state of the sensor."""
         return self._state
 
-    @asyncio.coroutine
-    def async_update(self):
+    async def async_update(self):
         """Retrieve latest state."""
         _LOGGER.debug("Updating Room sensor: %s", self._sensor)
         temp = self._eight.room_temperature()
```

Comment:
Does this method call do any I/O? If it does and it's not awaited properly, you're going to lock the event loop.

Reply:
It does not.  All the I/O happens in the '''async_update_heat_data''' and '''async_update_user_data''' functions in the master component file.  You can see the library room_temperature() function here: https://github.com/mezz64/pyEight/blob/d6c8e351acb9e43fd88be045d23ea4c6111bc8ff/pyeight/eight.py#L172